go/tools/bzltestutil/xml.go

```diff
@@ -102,6 +107,38 @@ func json2xml(r io.Reader, pkgName string) ([]byte, error) {
 				c.state = s
 			}
 		case "output":
+			trimmedOutput := strings.TrimSpace(e.Output)
+			if strings.HasPrefix(trimmedOutput, timeoutPanicPrefix) {
+				inTimeoutSection = true
+				// the final "fail" action with "Elapsed" may not appear. If not, using the timeout setting as
+				// pkgDuration; if it appears, it will override pkgDuration.
+				if duration, err := time.ParseDuration(strings.TrimSpace(trimmedOutput[len(timeoutPanicPrefix):])); err == nil {
+					seconds := duration.Seconds()
+					pkgDuration = &seconds
+				}
+				continue
+			}
+			if inTimeoutSection && strings.HasPrefix(trimmedOutput, "running tests:") {
+				inRunningTestSection = true
+				continue
+			}
+			if inRunningTestSection {
+				// looking for something like "TestReport/test_3 (2s)"
+				parts := strings.Fields(e.Output)
```

Comment:
Can subtest names contain spaces? We could split on the last space (or multiple spaces) instead.

Reply:
Spaces are allowed, but Go replaced them with `_`: https://github.com/golang/go/blob/215de81513286c010951624243c2923f7dc79675/src/testing/match.go#L282

In fact the test name here is called "test 3" in the code, but it became "test_3" from the report.

The error message is basically free-formed, so I am trying to be restrictive here, and avoid parsing wrong parts of the message. I set up 3 gates to trigger the logic:

1. look for "panic: test timed out after" first
2. look for "running tests" next
3. look for something like "TestReport/test_3 (2s)"

There is a lot of assumption on the implementation details in the testing package, which could change. But I would rather this parsing logic to stop working instead of updating wrong parts of the xml